src/App.tsx

```diff
@@ -15,71 +16,100 @@ export const goods = [
+      : this.setState({ selectedGood: good })
+  );
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  handleClick = () => {
+    this.setState({ selectedGood: '' });
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  render() {
+    const { selectedGood } = this.state;
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+    return (
+      <main className="section container">
+        {
+          selectedGood
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+            ? (
+              <h1 className="title is-flex is-align-items-center">
+                {`${selectedGood} is selected`}
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+                {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+                <button
+                  data-cy="ClearButton"
+                  type="button"
+                  className="delete ml-3"
+                  onClick={this.handleClick}
+                />
+              </h1>
+            )
+            : (<h1 className="title">No goods selected</h1>)
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+        }
+
+        <table className="table">
+          {goods.map(good => {
+            const rightTarget = selectedGood === good;
+
+            return (
+
+              <tbody>
+                <tr
+                  key={good}
+                  data-cy="Good"
+                  className={classNames(
+                    '',
+                    { 'has-background-success-light': rightTarget },
+                  )}
+                >
+                  <td>
+                    <button
+                      data-cy={
+                        rightTarget
+                          ? 'RemoveButton'
+                          : 'AddButton'
+                      }
+                      type="button"
+                      className={
+                        classNames(
+                          'button',
+                          { 'is-info': rightTarget },
+                        )
+                      }
+                      onClick={() => {
```

Comment:
checklist #2

Reply:
If your method get event as argument or don't get any argument, you don't need to create new arrow function. Just pass your method.
 
 but in this case i have 2 arguments
 so I can use new arrow function?
 
 sorry,I didn't understand your comment